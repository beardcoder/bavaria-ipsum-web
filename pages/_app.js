import { createGlobalStyle, ThemeProvider } from 'styled-components'
import 'react-toastify/dist/ReactToastify.css'
import 'normalize.css'
import '@fontsource/rubik'
import '@fontsource/germania-one'
import Head from 'next/head'

const GlobalStyle = createGlobalStyle`
  html,
  body {
    font-family: Rubik, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
      Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
      sans-serif;
      line-height: 1.2
  }

  p {
    line-height: 1.5
  }

  p:first-child {
    margin-top: 0;
  }

  p:last-child {
    margin-bottom: 0;
  }
`

const theme = {
  colors: {
    primary: '#0070f3',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export default function App({ Component, pageProps }) {
  return (
    <>
      <Head>
        <link
          rel="apple-touch-icon"
          sizes="180x180"
          href="/apple-touch-icon.png"
        />
        <link
          rel="icon"
          type="image/png"
          sizes="32x32"
          href="/favicon-32x32.png"
        />
        <link
          rel="icon"
          type="image/png"
          sizes="16x16"
          href="/favicon-16x16.png"
        />
        <link rel="manifest" href="/site.webmanifest" />
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5" />
        <meta name="msapplication-TileColor" content="#da532c" />
        <meta name="theme-color" content="#ffffff" />
      </Head>
      <GlobalStyle />
      <ThemeProvider theme={theme}>
        <Component {...pageProps} />
      </ThemeProvider>
    </>
  )
}
