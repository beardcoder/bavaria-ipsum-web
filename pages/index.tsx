import Head from 'next/head'
import { useState } from 'react'
import Blindtext from '../components/Blindtext'
import NoSSR from '@mpth/react-no-ssr'
import Slider from '../components/Slider'
import Controls from '../components/Controls'
import Header from '../components/Header'
import styled from 'styled-components'
import Footer from '../components/Footer'
import { ToastContainer } from 'react-toastify'

const Title = styled.h1`
  text-indent: -99999px;
  margin-top: 40px;
  overflow: hidden;
  width: 411px;
  height: 184px;
  background-image: url('/title.svg');
`

type Pros = {
  test?: boolean
}

export const Home = ({ test = false }: Pros): JSX.Element => {
  const [lenght, setLenght] = useState(3)
  const [paragraphs, setParagraphs] = useState(4)
  return (
    <div className="container">
      <Head>
        <title>Bavaria Ipsum</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Header>
        <Title>Bavaria Ipsum</Title>
        <Controls>
          <Slider
            label="Wia lang solls sei?"
            value={lenght}
            id="lenght"
            min={1}
            max={7}
            onChange={(value) => setLenght(value ?? 4)}
          ></Slider>
          <Slider
            label="Wia vui mechts ham?"
            value={paragraphs}
            id="paragraphs"
            min={1}
            max={10}
            onChange={(value) => setParagraphs(value ?? 4)}
          ></Slider>
        </Controls>
      </Header>
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
      <main>
        <div className="blindtext">
          {!test && (
            <NoSSR>
              <Blindtext lenght={lenght} paragraphs={paragraphs}></Blindtext>
            </NoSSR>
          )}
        </div>
      </main>

      <Footer></Footer>

      <style jsx>{`
        footer {
          width: 100%;
          height: 100px;
          border-top: 1px solid #eaeaea;
          display: flex;
          justify-content: center;
          align-items: center;
        }

        footer img {
          margin-left: 0.5rem;
        }

        footer a {
          display: flex;
          text-decoration: none;
          color: #121212;
          justify-content: center;
          align-items: center;
        }
      `}</style>

      <style jsx global>{`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto,
            Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
            sans-serif;
        }

        * {
          box-sizing: border-box;
        }

        p:first-child {
          margin-top: 0;
        }

        p:last-child {
          margin-bottom: 0;
        }
      `}</style>
    </div>
  )
}

export default Home
