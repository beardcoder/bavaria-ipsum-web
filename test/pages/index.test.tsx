import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home test />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
