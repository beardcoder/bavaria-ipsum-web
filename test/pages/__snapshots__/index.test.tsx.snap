// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-3115018971 container"
  >
    <header
      class="Header__Wrapper-sc-82l316-0 hFEnBr"
    >
      <h1
        class="pages__Title-sc-1rar0h2-0 dFSGiB"
      >
        Bavaria Ipsum
      </h1>
      <div
        class="Controls__Wrapper-sc-21ji11-0 etClar"
      >
        <div
          class="Slider__Wrapper-sc-182smj2-0 cLQGzt"
        >
          <label
            class="Slider__Label-sc-182smj2-1 dupCEA"
            for="lenght"
          >
            Wia lang solls sei?
          </label>
          <div
            class="Slider__InputWrapper-sc-182smj2-2 hYuoc"
          >
            <div
              class="Slider__InputValue-sc-182smj2-3 dhYaLC"
            >
              3
            </div>
            <div
              class="css-10fcxrz-Slider"
            >
              <div
                class="css-cdyzi2-Slider"
                style="width: 33.33333333333333%;"
              />
              <div
                style="position: absolute; transform: translate(-50%, -50%); left: 33.33333333333333%; top: 50%;"
              >
                <div
                  class="css-2byy62-Slider"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="Slider__Wrapper-sc-182smj2-0 cLQGzt"
        >
          <label
            class="Slider__Label-sc-182smj2-1 dupCEA"
            for="paragraphs"
          >
            Wia vui mechts ham?
          </label>
          <div
            class="Slider__InputWrapper-sc-182smj2-2 hYuoc"
          >
            <div
              class="Slider__InputValue-sc-182smj2-3 dhYaLC"
            >
              4
            </div>
            <div
              class="css-10fcxrz-Slider"
            >
              <div
                class="css-cdyzi2-Slider"
                style="width: 33.33333333333333%;"
              />
              <div
                style="position: absolute; transform: translate(-50%, -50%); left: 33.33333333333333%; top: 50%;"
              >
                <div
                  class="css-2byy62-Slider"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="Toastify"
    />
    <main
      class="jsx-3115018971"
    >
      <div
        class="jsx-3115018971 blindtext"
      />
    </main>
    <footer
      class="Footer__Wrapper-sc-13sxc4i-0 fVzfEB"
    >
      <a
        class="Footer__Link-sc-13sxc4i-1 fPsXMZ"
        href="https://creativeworkspace.de"
        rel="noopener noreferrer"
        target="_blank"
      >
        Desinged and Developed with ðŸ¦„ and â˜• by Markus Sommer
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
